internal/storage/index.go

```diff
@@ -114,3 +114,110 @@ func LoadPackagesAndCursorFromIndex(ctx context.Context, logger *zap.Logger, sto
 	}
 	return anIndex, storageCursor.Current, nil
 }
+
+func loadSearchIndexAllBatches(ctx context.Context, logger *zap.Logger, storageClient *storage.Client, bucketName, rootStoragePath string, aCursor cursor, batchSize int, process func(*packages.Packages) error) error {
```

Comment:
Currently duplicated the function. Not sure how to avoid this duplication now to not affect the current storage indexer.